### ISSUE TITLE
Dynamically Change Navigation Mode

### ISSUE THREAD

**Issue by ChristianM92** [madoar/ng2-archwizard#70]
So my application starts off the navigation mode on strict. Depending on what my function returns, I may want to set the navigation mode to free. I have tried to change it dynamically and the wizard doesn't seem to respond correctly. Is this functionality possible?

**Comment by madoar** [comment]
What do you mean with function? Is this function called only once, during initialization, or at a later time, after the user has entered some input?
Currently the navigation mode is set only during the initialization of the page. 
To change it later on, it would be required to implement an `ngUpdate` method, inside `WizardComponent`, which detects changes to the `navigationMode` and builds based on these changes a new `NavigationMode` instance.
@ChristianM92 if you want, fell free to open a new PR with these changes.

**Comment by ChristianM92** [comment]
The function is called at a later time after some user input. Thanks for the feedback!

**Comment by ChristianM92** [comment]
@madoar So I am interested in making the changes and creating the pull request. But this is my first time contributing to a project like this. How do I go about with testing and doing this properly? I don't wanna mess anything up.

**Comment by madoar** [comment]
@ChristianM92 nice to hear, that you want to give this a try :)

For testing I would suggest, that you take a look at other tests as a reference. All test files are named `*.spec.ts`, i.e. if you're making some changes inside `wizard.component.ts`, the related tests should most likely be contained in `wizard.component.spec.ts`.
If you require some help writing the tests I would suggest, that you take a look at https://angular.io/guide/testing.
To run the tests, you just need to execute `npm test` inside a terminal inside the `ng2-archwizard` directory.

About how doing this properly:
I believe it's best if you just start by forking the project and implementing your solution for the problem. After you've finished implemented a solution, or when you stumble over an error you don't know how to solve, you can just create a PR. 
Afterwards I will take a look at it, and give you some comments. If the PR is complete, and I can't think of anything significant to improve I'll merge it :)

If you've any questions feel free to ask!

**Comment by kamlekar** [comment]
I have the same requirement. Is this been / being implemented? @ChristianM92  @madoar

**Comment by ChristianM92** [comment]
@kamlekar @madoar Other priorities on my project came up so I wasn't able to focus on implementing this feature. Hopefully someone else can add this feature.

**Comment by madoar** [comment]
I don't have time to implement this feature currently too.
But if anyone wants to give it a try, feel free to do so!
